### ISSUE TITLE
DCFileChooser does not show hidden files/directories on Linux

### ISSUE THREAD

**Issue by jhorcicka** [datacleaner/DataCleaner#1421]
On Linux DCFileChooser does not allow to go into hidden directories. Or does it? Is there some special configuration for that?

In development mode DATACLEANER_HOME is /home/user/.datacleaner which basically means that user can not open/save a job from/to default directories (jobs). In case of a new datastore default directory is already set to DATACLEANER_HOME but when a user goes "up" then (s)he can not go back. 

This can be fixed in DCFileChooser constructor by: 
```
setFileHidingEnabled(false);
```
But that would be for all cases on all operating systems which might not be the best solution. 

Another approach might be: 
```
if (System.getProperty("os.name").equals("Linux")) {
  setFileHidingEnabled(false);
}
```

Or there might be a configuration for this (in properties? in options dialog?). 

What do you suggest, @LosD or @kaspersorensen? 

![dc-home-dialog](https://cloud.githubusercontent.com/assets/13213915/16555641/fa4fc352-41d5-11e6-89a1-160897d222aa.png)
![open-dialog](https://cloud.githubusercontent.com/assets/13213915/16555642/fa56aa6e-41d5-11e6-9aee-c232962689fc.png)

**Comment by kaspersorensen** [comment]
Hmm I think the if-block is a bad solution. Probably the best solution would be not to have the dot in the folder name in the first place! It's also always creating quite a lot of issues on Windows machines to have such a folder name.

**Comment by jhorcicka** [comment]
A workaround is to type the directory name into "File name". 

![workaround](https://cloud.githubusercontent.com/assets/13213915/16576051/e1e9b676-428c-11e6-9b62-b6f5299b105b.png)

**Comment by LosD** [comment]
I totally agree with @kaspersor, but I think we should wait with that until we have a better strategy for it all. Soon we'll (probably) see other changes to configuration files and job files when we try to make jobs more independent of data stores, so I think we should do changes to folder and file names at the same time, so we won't have more layers of compatibility utils.

**Comment by jhorcicka** [comment]
OK, let's wait for the right time to do it. Thanks for your comments, guys.
